account_invoice_line_salesperson/models/account_invoice_line.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2021 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    user_id = fields.Many2one(
+        "res.users", "Sales Person", readonly=True, compute="_compute_user_id",
+    )
+
+    @api.multi
+    def _compute_user_id(self):
+        for line in self:
+            order = self.env["sale.order"].search([("name", "=", self.sale_order_name)])
+            if order:
+                user = self.env["res.users"].search([("id", "=", order.user_id.id)])
+        if user:
+            line.user_id = user
+        else:
+            line.user_id = line.create_uid
```

Comment:
```suggestion
    def _compute_salesman_id(self):
        for line in self:
            if line.sale_line_ids:
                line.salesman_id = line.sale_line_ids[0].salesman_id
            else:
                line.salesman_id = line.invoice_id.user_id
```

Reply:
Thank you, I edited and it works as expected.